test/tests/functional/pbs_hook_execjob_prologue.py

```diff
@@ -0,0 +1,234 @@
+        self.server.manager(MGR_CMD_SET, NODE, attr, id=self.hostC)
+
+        j = Job(TEST_USER)
+        attr = {'Resource_List.select': '3:ncpus=1',
+                'Resource_List.place': 'scatter',
+                'Resource_List.walltime': 30}
+        j.set_attributes(attr)
+        j.set_sleep_time(10)
+        jid = self.server.submit(j)
+
+        rv = self.momB.log_match("Job;%s;JOIN_JOB as node" % (jid), n=100,
+                                 max_attempts=10, interval=2)
+        self.assertTrue(rv)
+        rv = self.momC.log_match("Job;%s;JOIN_JOB as node" % (jid), n=100,
+                                 max_attempts=10, interval=2)
+        self.assertTrue(rv)
+        rv = self.momA.log_match("Job;%s;executed prologue hook" % (jid,),
+                                 n=100, max_attempts=10, interval=2)
+        self.assertTrue(rv)
+        rv = self.momB.log_match("Job;%s;executed prologue hook" % (jid,),
+                                 n=100, max_attempts=10, interval=2)
+        self.assertTrue(rv)
+        rv = self.momC.log_match("Job;%s;executed prologue hook" % (jid,),
+                                 n=100, max_attempts=10, interval=2)
+        self.assertTrue(rv)
+
+
+    def test_prologue_internal_error_no_fail_action(self):
+        """
+        Test a prologue hook with an internal error and no fail_action.
+        """
+        hook_name = "prologue_exception"
+        # multi-line string format using tuple
+        hook_body = ("import pbs\n"
+                     "e = pbs.event()\n"
+                     "x\n")
+
+        attr = {'event': 'execjob_prologue',
+                'enabled': 'True'}
+        self.server.create_import_hook(hook_name, attr, hook_body)
+
+        j = Job(TEST_USER)
+        attr = {'Resource_List.select': 'vnode={}'.format(self.hostA),
+                'Resource_List.walltime': 30}
+        j.set_attributes(attr)
+        j.set_sleep_time(10)
+        jid = self.server.submit(j)
+
+        self.server.expect(NODE, {'state': 'free'}, id=self.hostA, interval=4)
+
+
+    def test_prologue_internal_error_offline_vnodes(self):
+        """
+        Test a prologue hook with an internal error and
+        fail_action=offline_vnodes.
+        """
+        attr = {'resources_available.mem': '2gb',
+                'resources_available.ncpus': '1'}
+        self.server.create_vnodes(self.hostC, attr, 2, self.momC, delall=True)
+        self.server.manager(MGR_CMD_SET, NODE, attr, id=self.hostC)
+
+        hook_name = "prologue_exception"
+        # multi-line string format using tuple
+        hook_body = ("import pbs\n"
+                     "e = pbs.event()\n"
+                     "x\n")
+        attr = {'event': 'execjob_prologue',
+                'enabled': 'True',
+                'fail_action': 'offline_vnodes'}
+        self.server.create_import_hook(hook_name, attr, hook_body)
+
+        j = Job(TEST_USER)
+        attr = {'Resource_List.select': 'vnode={}[0]'.format(self.hostC),
```

Comment:
Same as above with format().  I'd rather see you use the % method (same with the rest of the file).